contracts/libs/Actions.sol

```diff
@@ -8,5 +8,80 @@ pragma solidity 0.6.10;
+        MintShortOption,
+        BurnShortOption,
+        DepositLongOption,
+        WithdrawLongOption,
+        DepositCollateral,
+        WithdrawCollateral,
+        Exercise,
+        Call
+    }
 
+    struct ActionArgs {
+        /// @notice The type of action that is being performed on the system
+        ActionType actionType;
+        /// @notice The address of the account owner
+        address owner;
+        /// @notice The address which we move assets from or to (depending on the action type)
+        address sender;
+        /// @notice The asset that is to be transfered
+        address asset;
+        /// @notice The address of the vault that is to be modified (if any)
+        uint256 vaultId;
+        /// @notice The amount of asset that is to be transfered
+        uint256 amount;
+        /**
+         * @notice Each vault can hold multiple short / long / collateral assets. In this version, we are restricting the scope to only 1 of each.
+         * In future versions this would be the index of the short / long / collateral asset that needs to be modified.
+         */
+        uint256 index;
+        /// @notice Any other data that needs to be passed in for arbitrary function calls
+        bytes data;
+    }
+
+    struct DepositArgs {
+        /// @notice The address of the account owner
+        address owner;
+        /// @notice The address of the vault to which the asset will be added
+        uint256 vaultId;
+        /// @notice The address from which we transfer the asset
+        address from;
+        /// @notice The asset that is to be deposited
+        address asset;
+        /**
+         * @notice Each vault can hold multiple short / long / collateral assets. In this version, we are restricting the scope to only 1 of each.
+         * In future versions this would be the index of the short / long / collateral asset that needs to be modified.
+         */
+        uint256 index;
+        /// @notice The amount of asset that is to be transfered
+        uint256 amount;
+    }
+
+    /**
+     * @notice Parses the passed in action argmuents to get the argmuents for a deposit action
+     * @param _args The general action arguments structure
+     * @return The arguments for a deposit action
+     */
+    function _parseDepositArgs(ActionArgs memory _args) internal returns (DepositArgs memory) {
+        require(
+            (uint256(_args.actionType) == uint256(ActionType.DepositLongOption)) ||
+                (uint256(_args.actionType) == uint256(ActionType.DepositCollateral)),
+            "Actions: can only parse arguments for deposit actions"
+        );
+        require(_args.owner != address(0), "Actions: cannot deposit to an invalid account");
+
+        DepositArgs memory args;
+        args.owner = _args.owner;
+        args.vaultId = _args.vaultId;
+        args.from = _args.sender;
+        args.asset = _args.asset;
+        args.index = _args.index;
+        args.amount = _args.amount;
+
```

Comment:
Would [this syntax from dydx](https://github.com/dydxprotocol/solo/blob/3b6db45cf109fa2cf272843c4192a7d892ad30f4/contracts/protocol/lib/Actions.sol#L265) be more readable?
```
return DepositArgs({
            amount: args.amount,
            account: accounts[args.accountId],
            market: args.primaryMarketId,
            from: args.otherAddress
        });
```

Reply:
hmm that changes how we have been doing return syntax so far. @haythem96 thoughts?